Makefile

```diff
@@ -11,6 +11,8 @@ INSTALL = install
 INSTALL_DIR = $(INSTALL) -m755 -d
 INSTALL_DATA = $(INSTALL) -m644
 INSTALL_PROGRAM = $(INSTALL) -m755
+GIT_REVISION = $(shell (git rev-parse HEAD 2> /dev/null) || (cat .git/HEAD 2> /dev/null | head -c 40))
```

Comment:
the commands are different, before it was only `git rev-parse HEAD`, but I have mimicked how we do it in the System.get_revision/0 function.

One important point I forgot to mention in the PR is that in the current MakeFile, `git` is a requirement if we want to build docs (when it should not), and if git is not installed, Makefile will complain.
```sh
/bin/sh: 1: git: not found
/bin/sh: 1: git: not found
```

With these changes, it will just fail silently.